src/types/response.rs

```diff
@@ -156,4 +156,14 @@ impl PyResponse {
             .into_py(py);
         Ok(())
     }
+
+    pub fn set_cookie(&mut self, py: Python, key: &str, value: &str) -> PyResult<()> {
+        let kwargs = vec![("key", key), ("value", value)].into_py_dict(py);
+
+        if let Err(e) = self.headers.call_method(py, "append", (), Some(kwargs)) {
```

Comment:
@VishnuSanal , calling methods in py handlers is usually slower. So it is preferred to alter rust handlers directly and then postpone the conversion until the end, if possible

Reply:
@sansyrox hi, I can't find another way to do this. I have removed the use of `pydict` now. please check. 

when using `headers.append` (as you mentioned in your code) leads me to the following error.

```
error[E0599]: no method named `append` found for struct `pyo3::Py` in the current scope
   --> src/types/response.rs:161:22
    |
161 |         self.headers.append(key.to_string(), value.to_string());
    |                      ^^^^^^ method not found in `Py<Headers>`

```